### PR TITLE
Add missing error metrics and stabilize Infinite Tracing integration tests 

### PR DESCRIPTION
**Pull request by JcolemanNR**
Thank you for submitting a pull request.  Please review our [contributing guidelines](/CONTRIBUTING.md) and [code of conduct](https://opensource.newrelic.com/code-of-conduct/).

## Description
Adds missing error supportability metric generation to gRPC response streams.

Changes the log lines that are waited on in Infinite Tracing integration tests to allow more deterministic test results.

Introduces a new Integration test log helper to wait on captured count totals for log line regexes.

# Author Checklist
- [x] Unit tests, Integration tests, and Unbounded tests completed
- [x] Performance testing completed with satisfactory results (if required)
- [x] [Agent Changelog](/src/Agent/CHANGELOG.md) or [Lambda Agent changelog](/src/AwsLambda/CHANGELOG.md) updated 

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)
- [ ] Review Changelog

### CODE REVIEW

File: tests/Agent/IntegrationTests/IntegrationTestHelpers/AgentLogBase.cs [hunk 1]
```diff
@@ -62,7 +62,8 @@ public abstract class AgentLogBase
         public const string AttemptReconnectLogLineRegex = InfoLogLinePrefixRegex + "Will attempt to reconnect in \\d{2,3} seconds";
 
         // Infinite trace
-        public const string SpanStreamingSuccessLogLineRegex = FinestLogLinePrefixRegex + @"SpanStreamingService: consumer \d+ - Attempting to send (\d+) item\(s\) - Success";
+        public const string SpanStreamingSuccessfullySentLogLineRegex = FinestLogLinePrefixRegex + @"SpanStreamingService: consumer \d+ - Attempting to send (\d+) item\(s\) - Success";
+        public const string SpanStreamingSuccessfullyProcessedByServerResponseLogLineRegex = FinestLogLinePrefixRegex + @"SpanStreamingService: consumer \d+ - Received gRPC Server response: (\d+)";
```
Comment (JcolemanNR) [inline tests/Agent/IntegrationTests/IntegrationTestHelpers/AgentLogBase.cs:66]: NOTE: This regex will need to be reconciled with the changes in @jaffinito's PR

File: tests/Agent/IntegrationTests/IntegrationTests/InfiniteTracing/InfiniteTracingTests.cs [hunk 2]
```diff
@@ -21,17 +21,16 @@ public abstract class InfiniteTracingTestsBase<TFixture> : NewRelicIntegrationTe
...
             _fixture = fixture;
-            _fixture.SetTimeout(System.TimeSpan.FromMinutes(2));
+            _fixture.SetTimeout(TimeSpan.FromMinutes(3));
             _fixture.TestLogger = output;
 
             _fixture.AddCommand($"InfiniteTracingTester StartAgent");
 
-            _fixture.AddCommand("RootCommands DelaySeconds 5"); // give the agent time to warm up
+            // Give the agent time to warm up... If we send a span too soon, it will be sent via DT (span_event_data) instead of 8T (gRPC)
+            _fixture.AddCommand("RootCommands DelaySeconds 15");
```
Comment (JcolemanNR) [inline tests/Agent/IntegrationTests/IntegrationTests/InfiniteTracing/InfiniteTracingTests.cs:30]: Feature request: We need an elegant way to wait for log lines interleaved with the ConsoleMF commands array

File: src/Agent/NewRelic/Agent/Core/DataTransport/DataStreamingService.cs [hunk 3]
```diff
@@ -544,8 +550,9 @@ private bool CreateChannel(CancellationToken cancellationToken)
                 }
                 catch (Exception ex)
                 {
+                    // TODO: I don't think this is actually a response error...
+                    //RecordResponseError();
```
Comment (nrcventura) [inline src/Agent/NewRelic/Agent/Core/DataTransport/DataStreamingService.cs:554]: Did you mean to include changes like this with the PR to make the integration tests more stable?

Comment (nrcventura) [reply src/Agent/NewRelic/Agent/Core/DataTransport/DataStreamingService.cs:554]: The call to `_grpcWrapper.CreateChannel` in the legacy library triggers a connection attempt which can result in networking errors or receiving a response from the server (such as grpc unavailable). So I don't think that we want to remove this code.

Comment (JcolemanNR) [reply src/Agent/NewRelic/Agent/Core/DataTransport/DataStreamingService.cs]: This has been re-added (along with the others I disabled for science)

**Review by tippmar-nr — DISMISSED** [review]

**Review by nrcventura — APPROVED** [review]

**Review by nrcventura — APPROVED** [review]

**Review by jaffinito — APPROVED** [review]
Nice to have more supportability metrics around this.  Excellent!

**Pull request merged**